docs/source/preview_custom_datasets.md

```diff
@@ -0,0 +1,24 @@
+# Extend Preview to Custom Datasets
+
+When creating a custom dataset, if you wish to enable data preview for that dataset, you must implement a `preview()` function within the custom dataset class. Kedro-viz currently supports previewing Tables, Plotly charts, Images, and JSON objects.
+
+The return type of the `preview()` function should match one of the following types, as defined in the `kedro-datasets` source code (_typing.py file):
+
+```python
+TablePreview = NewType("TablePreview", dict)
+ImagePreview = NewType("ImagePreview", bytes)
+PlotlyPreview = NewType("PlotlyPreview", dict)
+JSONPreview = NewType("JSONPreview", dict)
+```
```

Comment:
Can we show an example for each type? This help users to link what are they expecting to see in the UI.

https://clear.ml/docs/latest/docs/guides/reporting/plotly_reporting